kernel/virtio/virtio_net.c

```diff
@@ -203,38 +203,40 @@ static void recv_setup(void)
 int virtio_net_xmit_packet(void *data, int len)
 {
     struct virtq_desc *desc;
-    struct virtq_avail *avail;
     int dbg = 0;
 
-    if (((xmit_next_avail + 1) % xmitq.num) ==
-        (xmit_last_used % xmitq.num)) {
+    if (((xmit_next_avail + 2) % xmitq.num) ==
+        ((xmit_last_used * 2) % xmitq.num)) {
```

Comment:
Not using this check anymore. The new version keeps track of free descriptors explicitly (in virtq.num_avail).